virttest/utils_net.py

```diff
@@ -3705,6 +3706,32 @@ def _process_output(cmd, reg_pattern):
                                "mac %s" % mac_address)
 
 
+def get_linux_iface_info(mac, session=None):
+    """
+    Get host/guest info of certain interface with given mac address.
+
+    :param mac: string type, mac address of given interface
+    :param session: session of given vm, default to None
+    :return: dict-type info of interface, None if not get any
+    """
+    ip_cmd = 'ip -json a'
+
+    try:
+        if session:
+            ip_output_str = session.cmd_output(ip_cmd).strip()
+        else:
+            ip_output_str = process.run(ip_cmd).stdout_text.strip()
+        ip_info = json.loads(ip_output_str)
+        logging.debug('interfaces inside vm:\n %s', ip_info)
+    except Exception:
+        logging.error('Failed to get interfaces inside vm.')
```

Comment:
Is it better to log the detailed exception message, too?

Reply:
Of course, nice catch!